### PR TITLE
Feature/maintenance

### PR DESCRIPTION
**Pull request by dionisislolas**
### What

Upgrade dependencies.
Upgrade go version.
Update linter.
Update env vars to match readme.
Update swagger specs.

### How to review

Changes make sense. Test passes.

### Who can review

Anyone.

### CODE REVIEW

**Review by lindenmckenzie — COMMENTED** [review]
This is very cheeky, but could you upgrade the version of the CI lint image to 1.53.2 and run the checks again - you'll probably need to switch off `depguard`

File: api/api.go [hunk 1]
```diff
@@ -18,10 +18,10 @@ import (
 )
 
 var (
-	createPermission = auth.Permissions{Create: true}
+	// createPermission = auth.Permissions{Create: true}
 	readPermission   = auth.Permissions{Read: true}
 	updatePermission = auth.Permissions{Update: true}
-	deletePermission = auth.Permissions{Delete: true}
+	// deletePermission = auth.Permissions{Delete: true}
 )
```
Comment (lindenmckenzie) [inline api/api.go:26]: Why are these lined commented out?

File: api/api.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func Setup(ctx context.Context, cfg *config.Config, router *mux.Router, dataStor
 }
 
 // enablePublicEndpoints register only the public GET endpoints.
```
Comment (lindenmckenzie) [inline api/api.go:68]: Rather than blanking these, might be worth removing them as parameters and updating their reference functions.

File: ci/build.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.19.2
+    tag: 1.20.5
```
Comment (lindenmckenzie) [inline ci/build.yml:9]: You might have issues with 1.20.5 - see how it goes.

File: config/config.go [hunk 4]
```diff
@@ -42,7 +42,7 @@ func Get() (*Config, error) {
 		BindAddr:                   "localhost:25300",
 		EnablePermissionsAuth:      false,
 		EnablePrivateEndpoints:     false,
-		GracefulShutdownTimeout:    10 * time.Second,
+		GracefulShutdownTimeout:    5 * time.Second,
 		HealthCheckCriticalTimeout: 90 * time.Second,
```
Comment (lindenmckenzie) [inline config/config.go:46]: Why this change?

File: go.mod [hunk 5]
```diff
@@ -1,75 +1,74 @@
...
 	github.com/kelseyhightower/envconfig v1.4.0
 	github.com/pkg/errors v0.9.1
-	github.com/smartystreets/goconvey v1.7.2
-	go.mongodb.org/mongo-driver v1.9.1
+	github.com/smartystreets/goconvey v1.8.0
+	github.com/stretchr/testify v1.8.4
+	go.mongodb.org/mongo-driver v1.11.7
 )
 
 require github.com/hashicorp/go-uuid v1.0.3 // indirect
```
Comment (lindenmckenzie) [inline go.mod:29]: why's this one by itself? did you do a `go mod tidy`?

File: main.go [hunk 6]
```diff
@@ -36,7 +36,6 @@ func main() {
 
 	if err := run(ctx); err != nil {
 		log.Fatal(ctx, "fatal runtime error", err)
-		os.Exit(1)
```
Comment (lindenmckenzie) [inline main.go:39]: It would be better to change the log level to `Error` rather than remove the os.Exit I think here.

File: swagger.yaml [hunk 7]
```diff
@@ -52,6 +52,23 @@ parameters:
     schema:
       $ref: "#/definitions/TopicUpdate"
 paths:
+  /topics/:
+    get:
+      tags:
+        - "Public"
+      summary: "Get a lists of topics"
+      description: "gets a public list of top level root topics"
```
Comment (lindenmckenzie) [inline swagger.yaml:60]: "Gets a public list..."

File: swagger.yaml [hunk 8]
```diff
@@ -52,6 +52,23 @@ parameters:
...
+      produces:
+        - "application/json"
+      responses:
+        200:
+          description: "Json object containing a list of all available topics"
```
Comment (lindenmckenzie) [inline swagger.yaml:65]: "...all available _root_ topics"

File: swagger.yaml [hunk 9]
```diff
@@ -52,6 +52,23 @@ parameters:
...
+          schema:
+            $ref: '#/definitions/ListOfTopics'
+        404:
+          description: "Topic not found"
```
Comment (lindenmckenzie) [inline swagger.yaml:69]: I don't think this is accurate but I'm not sure what to put instead. What causes it to give a 404? I'm not sure it can!

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
Few comments - I'm not sure about blanking inputs, it would probably be better to go and update the code inputting to them as well and remove it entirely.

File: api/api.go (hunk 1 above)
Comment (dionisislolas) [reply api/api.go]: I introduced a nolint comment for this to pass the make lint command. Is this ok ?

File: api/api.go (hunk 2 above)
Comment (dionisislolas) [reply api/api.go:69]: This is now addressed.

File: ci/build.yml (hunk 3 above)
Comment (dionisislolas) [reply ci/build.yml:9]: I left this as it is to see how it goes.

File: config/config.go (hunk 4 above)
Comment (dionisislolas) [reply config/config.go]: This is noe addressed.

File: go.mod (hunk 5 above)
Comment (dionisislolas) [reply go.mod]: For some reason go mod tidy does not remove it. Had to remove it by hand.

File: main.go (hunk 6 above)
Comment (dionisislolas) [reply main.go]: This is now addressed.

File: swagger.yaml (hunk 7 above)
Comment (dionisislolas) [reply swagger.yaml]: This is now addressed.

File: swagger.yaml (hunk 8 above)
Comment (dionisislolas) [reply swagger.yaml]: This is now addressed.

**Review by lindenmckenzie — APPROVED** [review]
